tests/kafkatest/tests/verifiable_consumer_test.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.utils.util import wait_until
+
+from kafkatest.tests.kafka_test import KafkaTest
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.verifiable_consumer import VerifiableConsumer
+from kafkatest.services.kafka import TopicPartition
+
+class VerifiableConsumerTest(KafkaTest):
```

Comment:
How close is this to the `ProduceConsumeValidateTest` class? It seems like there might be quite a bit of overlap, although I'm still looking through the rest of this patch to figure out how much of the tests that inherit from this actually depend on the VerifiableConsumer. The usage seems a bit odd, e.g. we have a method for allocating the consumer/producer, but they aren't saved as fields in the test? Which means that methods like `valid_assignment` need data passed in (`assignment`) which should just be extracted via the consumer that the test created?